### PR TITLE
OCM-274: Separate setup and teardown of rosa-terraform

### PR DESCRIPTION
**Pull request by osherdp**
Use the new makefile targets ``apply_folder`` and ``destroy_folder`` for setting up terraform modules and tearing them up.

This allows inserting testing steps (that only depend on a ``${SHARED_DIR}/kubeconfig`` file for their run) in between the ``pre`` and ``post`` phases.

### CODE REVIEW

**Comment by osherdp** [comment]
/pj-rehearse pull-ci-terraform-redhat-terraform-provider-ocm-OCM-274-e2e-presubmit

**Comment by osherdp** [comment]
/pj-rehearse pull-ci-terraform-redhat-terraform-provider-ocm-OCM-274-e2e-presubmit

**Comment by sagidayan** [comment]
/pj-rehearse pull-ci-terraform-redhat-terraform-provider-ocm-OCM-274-e2e-presubmit

**Comment by osherdp** [comment]
/pj-rehearse pull-ci-terraform-redhat-terraform-provider-ocm-OCM-274-e2e-presubmit

File: ci-operator/step-registry/rosa/terraform/aws/e2e/rosa-terraform-aws-e2e-workflow.yaml [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+    cluster_profile: aws-qe
+    allow_best_effort_post_steps: true
+    pre:
+      - ref: rosa-terraform-aws-e2e-setup
+    post:
+      - ref: rosa-terraform-aws-e2e-teardown
+    env:
+      OPENSHIFT_VERSION: "4.13"
+  documentation: |-
+    This workflow executes the common end-to-end test suite on a cluster provisioned by running assisted-installer on a packet server.
```
Comment (osherdp) [inline ci-operator/step-registry/rosa/terraform/aws/e2e/rosa-terraform-aws-e2e-workflow.yaml:13]: whoopsie, I'll change that

**Comment by osherdp** [comment]
/pj-rehearse pull-ci-terraform-redhat-terraform-provider-ocm-OCM-274-e2e-presubmit

**Comment by osherdp** [comment]
/pj-rehearse pull-ci-terraform-redhat-terraform-provider-ocm-OCM-274-e2e-presubmit

File: ci-operator/config/terraform-redhat/terraform-provider-ocm/terraform-redhat-terraform-provider-ocm-OCM-274.yaml [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+      namespace: ocp
+resources:
+  '*':
+    limits:
+      memory: 4Gi
+    requests:
+      cpu: 100m
+      memory: 200Mi
+tests:
+- as: e2e-presubmit
```
Comment (AsherShoshan) [inline ci-operator/config/terraform-redhat/terraform-provider-ocm/terraform-redhat-terraform-provider-ocm-OCM-274.yaml:23]: refinement of pre-submit are gone..

**Comment by osherdp** [comment]
/pj-rehearse pull-ci-terraform-redhat-terraform-provider-ocm-OCM-274-e2e-presubmit

**Comment by osherdp** [comment]
/pj-rehearse pull-ci-terraform-redhat-terraform-provider-ocm-OCM-274-e2e-presubmit

**Comment by osherdp** [comment]
/pj-rehearse pull-ci-terraform-redhat-terraform-provider-ocm-OCM-274-e2e-presubmit

**Comment by osherdp** [comment]
/pj-rehearse pull-ci-terraform-redhat-terraform-provider-ocm-OCM-274-e2e-presubmit

**Comment by osherdp** [comment]
/pj-rehearse pull-ci-terraform-redhat-terraform-provider-ocm-OCM-274-e2e-presubmit

**Comment by osherdp** [comment]
/pj-rehearse pull-ci-terraform-redhat-terraform-provider-ocm-OCM-274-e2e-presubmit

**Comment by osherdp** [comment]
/pj-rehearse pull-ci-terraform-redhat-terraform-provider-ocm-OCM-274-e2e-presubmit

File: ci-operator/config/terraform-redhat/terraform-provider-ocm/terraform-redhat-terraform-provider-ocm-OCM-274.yaml (hunk 2 above)
Comment (osherdp) [reply ci-operator/config/terraform-redhat/terraform-provider-ocm/terraform-redhat-terraform-provider-ocm-OCM-274.yaml:23]: I'll keep it in place before merging it to the ``main`` config

**Comment by osherdp** [comment]
/pj-rehearse pull-ci-terraform-redhat-terraform-provider-ocm-OCM-274-e2e-presubmit

**Comment by osherdp** [comment]
/pj-rehearse pull-ci-terraform-redhat-terraform-provider-ocm-OCM-274-e2e-presubmit

**Comment by osherdp** [comment]
/pj-rehearse pull-ci-terraform-redhat-terraform-provider-ocm-OCM-274-e2e-presubmit

**Comment by sagidayan** [comment]
/pj-rehearse pull-ci-terraform-redhat-terraform-provider-ocm-OCM-274-e2e-presubmit

**Comment by osherdp** [comment]
/pj-rehearse pull-ci-terraform-redhat-terraform-provider-ocm-OCM-274-e2e-presubmit

**Comment by sagidayan** [comment]
/pj-rehearse pull-ci-terraform-redhat-terraform-provider-ocm-OCM-274-e2e-presubmit

**Review by sagidayan — APPROVED** [review]
Great! Thanks 👏🏽

**Comment by AsherShoshan** [comment]
@osherdp Please remove the e2e/test folder

**Comment by osherdp** [comment]
> @osherdp Please remove the e2e/test folder

I removed it
```
$ tree ci-operator/step-registry/rosa/terraform/
ci-operator/step-registry/rosa/terraform/
├── e2e
│   ├── OWNERS
│   ├── setup
│   │   ├── OWNERS
│   │   ├── rosa-terraform-e2e-setup-commands.sh
│   │   ├── rosa-terraform-e2e-setup-ref.metadata.json
│   │   └── rosa-terraform-e2e-setup-ref.yaml
│   └── teardown
│       ├── OWNERS
│       ├── rosa-terraform-e2e-teardown-commands.sh
│       ├── rosa-terraform-e2e-teardown-ref.metadata.json
│       └── rosa-terraform-e2e-teardown-ref.yaml
└── OWNERS

4 directories, 10 files
```

**Pull request merged**